src/features/atomics.md

```diff
@@ -0,0 +1,202 @@
+---
+title: 'Atomics.wait, Atomics.notify, Atomics.waitAsync'
+author: '[Marja Hölttä](https://twitter.com/marjakh), a non-blocking blogger'
+avatars:
+  - marja-holtta
+date: 2020-09-14
+tags:
+  - ECMAScript
+  - ES2020
+description: 'The JavaScript nullish coalescing operator enables safer default expressions.'
+tweet: ''
+---
+
+# Atomics.wait, Atomics.notify, Atomics.waitAsync
+
+[`Atomics.wait`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/wait) and [`Atomics.notify`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/notify) are low-level synchronization primitives useful for implementing mutexes and other means of synchronization. However, since `Atomics.wait` is blocking, it’s not possible to call it on the main thread (trying to do so will throw a TypeError).
+
+Starting from Chrome 87, V8 supports a non-blocking version, [Atomics.waitAsync](https://github.com/tc39/proposal-atomics-wait-async/blob/master/PROPOSAL.md), which is also usable on the main thread.
+
+In this post, we explain how to use these low-level APIs to implement a mutex that works both synchronously (for worker threads) and asynchronously (for worker threads or the main thread).
+
+`Atomics.wait` and `Atomics.waitAsync` take the following parameters:
+
+- an `Int32Array` or `BigInt64Array` backed by a `SharedArrayBuffer`
+- index (valid within the array)
+- a value we expect to be present in the memory location pointed to by buffer and index
+- timeout in milliseconds (optional, defaults to `Infinity`)
+
+The return value of `Atomics.wait` is a string. If the memory location doesn’t contain the expected value, `Atomics.wait` returns immediately with the value `"not-equal"`. Otherwise, the thread is blocked until another thread calls `Atomics.notify` with the same memory location or the timeout is reached. In the former case, `Atomics.wait` returns the value `"ok"`, in the latter case, `Atomics.wait` returns the value `"timed-out"`.
+
+`Atomics.notify` takes the following parameters:
+
+- an `Int32Array` or `BigInt64Array` backed by a `SharedArrayBuffer`
+- index (valid within the array)
+- how many waiters to notify (optional, defaults to `Infinity`)
+
+It notifies the given amout of waiters, in FIFO order, waiting on the memory location described by (buffer, index). If there are several pending `Atomics.wait` calls or `Atomics.waitAsync` calls related to the same location, they are all in the same FIFO queue.
```

Comment:
Maybe worth naming params `buffer` and `index` in the list above to make a clear reference from this sentence (`(buffer, index)`).

Reply:
done